hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -311,6 +310,35 @@ public void onConfigurationChange(Configuration newConf) {
     if (authorize) {
       refreshAuthManager(newConf, new HBasePolicyProvider());
     }
+    refreshSlowLogConfiguration(newConf);
```

Comment:
it probably makes sense to move this into `initReconfigurable`. Then you can remove the assignments altogether from the constructor.

Reply:
since we don't have a guarantee that the constructors of RpcServer's extensions will call `initReconfigurable`, I wonder whether we should leave the assignments in the constructor to be safe

Reply:
ok, tbh im not sure why they chose to call initReconfigurable in the subclasses. I'm guessing it has something to do with calling it before initializing the scheduler.

Since that doesn't matter to us, I don't mind reverting back to your original calling location (in onConfigurationChange). It does seem weird to put a refresh method (which updates rather than sets) in an init method (which sets).